lightstep/trace_context.py

```diff
@@ -0,0 +1,311 @@
+                )
+
+            # https://www.w3.org/TR/2019/CR-trace-context-20190813/#trace-flags
+            raw_trace_flags = remainder_match.group("trace_flags")
+
+            trace_flags = []
+
+            for index, bit_flag in enumerate(
+                zip(
+                    bin(int(raw_trace_flags, 16))[2:].zfill(8),
+                    # https://www.w3.org/TR/2019/CR-trace-context-20190813/#other-flags
+                    # Flags can be added in the next list as the document
+                    # progresses and they get defined. This list represents the
+                    # 8 bits that are available in trace-flags and their
+                    # respective meaning.
+                    [None, None, None, None, None, None, None, "sampled"]
+                )
+            ):
+                bit, flag = bit_flag
+
+                if int(bit):
+                    if flag is None:
+                        warn("Invalid flag set at bit {}".format(index))
+                        trace_flags.append("0")
+
+                    else:
+                        trace_flags.append("1")
+
+                else:
+                    trace_flags.append("0")
+
+            trace_context_free_carrier["trace-flags"] = int(
+                "".join(trace_flags), 2
+            )
+
+            if tracestate is not None:
+
+                tracestate_dictionary = OrderedDict()
+
+                for counter, list_member in enumerate(tracestate.split(",")):
+                    # https://www.w3.org/TR/trace-context/#tracestate-header-field-values
+                    if counter > 31:
+                        _LOG.warning(
+                            "More than 32 list-members "
+                            "found in tracestate {}".format(tracestate)
+                        )
+                        break
+
+                    # https://www.w3.org/TR/trace-context/#tracestate-header-field-values
+                    if _BLANK.match(list_member):
+                        _LOG.debug(
+                            "Ignoring empty tracestate list-member"
+                        )
+                        continue
+
+                    key_value = _KEY_VALUE.match(list_member)
+
+                    if key_value is None:
+                        _LOG.warning(
+                            "Invalid key/value pair found: {}".format(
+                                key_value
+                            )
+                        )
+                        break
+
+                    key, value = key_value.groups()
+
+                    if key in tracestate_dictionary.keys():
+                        _LOG.warning(
+                            "Duplicate tracestate key found: {}".format(key)
+                        )
+                        break
+
+                    tracestate_dictionary[key] = value
+
+                else:
+                    trace_context_free_carrier["tracestate"] = ",".join(
```

Comment:
Minor change here: should use `_TRACESTATE`

Reply:
Fixed!